Задача: Lord of the Rings RPG.

piligrim-curious opened the pull request:
**Character**: Класс персонажа;
**Item**: Класс предметов;
**InventoryManager**: Класс операций.

piligrim-curious committed:
Задача: Lord of the Rings RPG.
Character: Класс персонажа;
Item: Класс предметов;
InventoryManager: Класс операций.

piligrim-curious committed:
Задача: Lord of the Rings RPG.
Character: Класс персонажа;
Item: Класс предметов;
InventoryManager: Класс операций.

SnurnicynKonstantin commented on src/main/java/school/faang/lord/of/the/rings/rpg/Character.java:
```diff
@@ -0,0 +1,13 @@
+
+import java.util.List;
+
+@Builder
```
Рас ты использовал Builder, то вот тебе подсказка о том, как можно создать объект с помощью него.
`Character cail = Character.builder().name("Cail").inventory(new ArrayList<>()).build();`
Ты можешь хоть с одним name создать объект, хоть с двумя полями. С помощью билдера не надо создавать конструкторы под разные вариации

piligrim-curious replied:
**Спасибо**), очень полезно, а то я голову ломал как создать объект класса с конкретными полями, без конструктора.

SnurnicynKonstantin commented on src/main/java/school/faang/lord/of/the/rings/rpg/InventoryManager.java:
```diff
@@ -0,0 +1,67 @@
+            throw new IllegalArgumentException("Character and filter cannot be null");
+        }
+        ListIterator<Item> iterator = inventory.listIterator();
+        while (iterator.hasNext()) {
```
А попробуй с помощью for сделать, вроде короче получится. Например
```
        for (int i = 0; i < inventory.size(); i++) {
            Item item = inventory.get(i);
            if (filter.test(item)) {
                inventory.set(i, updater.apply(item));
            }
        }
```

piligrim-curious committed:
Задача: Lord of the Rings RPG.
Character: Класс персонажа;
Item: Класс предметов;
InventoryManager: Класс операций.

Изменения:
(InventoryManager.java)
* Изменён метод updateItem();
* Изменена инициализация объекта Character.

SnurnicynKonstantin reviewed (approved):
Билдер освоен, супер)
Билдер часто используется
